software/contrib/egressus_melodiam.py

```diff
@@ -0,0 +1,1091 @@
+            oled.pixel(13, 31, 1)
+            oled.pixel(14, 31, 1)
+
+    """Update the screen only if something has changed. oled.show() hogs the processor and causes latency."""
+
+    def updateScreen(self):
+
+        # Only update if something has changed
+        if not self.screenRefreshNeeded:
+            return
+        # Clear screen
+        oled.fill(0)
+
+        # Selected output
+        oled.fill_rect(108, 0, 20, 9, 1)
+        oled.text(f"{self.selectedOutput + 1}", 115, 1, 0)
+
+        # Show division for selected output
+        number = self.outputDivisions[self.selectedOutput]
+        x = 111 if number >= 10 else 115
+        oled.text(f"{number}", x, 12, 1)
+
+        # Show wave for selected output
+        self.drawWave()
+
+        if self.unClockedMode:
+
+            if self.averageMsBetweenClocks != 0:
+                oled.text(f"{int(self.averageMsBetweenClocks * 2)} ms", 31, 1, 1)
+
+        else:
+
+            # Draw pattern length
+            row1 = ""
+            row2 = ""
+            row3 = ""
+            row4 = ""
+            if self.patternLength > 24:
+                # draw two rows
+                row1 = "........"
+                row2 = "........"
+                row3 = "........"
+                for i in range(24, self.patternLength):
+                    row4 = row4 + "."
+            elif self.patternLength > 16:
+                row1 = "........"
+                row2 = "........"
+                for i in range(16, self.patternLength):
+                    row3 = row3 + "."
+            elif self.patternLength > 8:
+                row1 = "........"
+                for i in range(8, self.patternLength):
+                    row2 = row2 + "."
+            else:
+                # draw one row
+                for i in range(self.patternLength):
+                    row1 = row1 + "."
+
+            xStart = 31
+            oled.text(row1, xStart, 0, 1)
+            oled.text(row2, xStart, 6, 1)
+            oled.text(row3, xStart, 12, 1)
+            oled.text(row4, xStart, 18, 1)
+
+        # Draw a visual cue for when a long button press has been detected
+        # and a new random pattern is being generated
+        if self.shreadedVis:
+            oled.fill_rect(0, 0, 20, 9, 1)
+            oled.pixel(2, 8, 0)
+            oled.pixel(2, 7, 0)
+            oled.pixel(2, 6, 0)
+            oled.pixel(3, 5, 0)
+            oled.pixel(3, 4, 0)
+            oled.pixel(4, 3, 0)
+            oled.pixel(5, 3, 0)
+            oled.pixel(6, 4, 0)
+            oled.pixel(7, 5, 0)
+            oled.pixel(8, 5, 0)
+            oled.pixel(9, 4, 0)
+            oled.pixel(9, 3, 0)
+            oled.pixel(9, 2, 0)
+            oled.pixel(10, 1, 0)
+            oled.pixel(11, 2, 0)
+            oled.pixel(12, 3, 0)
+            oled.pixel(12, 4, 0)
+            oled.pixel(13, 5, 0)
+            oled.pixel(13, 6, 0)
+            oled.pixel(14, 7, 0)
+            oled.pixel(15, 8, 0)

```

Comment:
[discussion]
Another instance where a bitmap and `.blit(...)` would probably be useful.

Reply:
Yep, agreed. Done!